app/models/user.rb

```diff
@@ -22,7 +23,15 @@ class User < ApplicationRecord
 
   has_many :comments
 
-  has_many :likes
+  has_many :likes, dependent: :destroy
+  has_many :liked_comments, through: :likes, source: :likable, source_type: 'Comment'
```

Comment:
I am not sure how this association will be used in the app later on so I will just revisit this again when that time comes to use it.